tools/empath-split.py

```diff
@@ -0,0 +1,233 @@
+  # map from it
+  with webassembly.Module(wasm) as module:
+    if not module.has_name_section():
+      exit_with_error('Name section does not eixst')
+    if not sourcemap:
+      if not emsymbolizer.get_sourceMappingURL_section(module):
+        exit_with_error('sourceMappingURL section does not exist')
+
+    funcs = module.get_functions()
+    func_names = module.get_function_names()
+    assert len(funcs) == len(func_names)
+
+    func_to_src = {}
+    src_to_funcs = {}
+
+    if not sourcemap:
+      sourcemap = module.get_sourceMappingURL()
+    sm = emsymbolizer.WasmSourceMap()
+    sm.parse(sourcemap)
+
+    for func_name, func in zip(func_names, funcs):
+      # From the last address, decrement the address by 1 until we find location
+      # info with source file information. The reason we do this is to reduce
+      # the probability of picking an address where another function is inlined
+      # into, picking the inlined function's source.
+      # We start from the end because it is simpler; it is harder to compute the
+      # first instruction's address, because there is a gap for local types
+      # between function offset and the first instruction.
+      addr = func.offset + func.size - 1
+      while addr > func.offset:
+        loc = sm.lookup(addr, func.offset)
+        # This means there is no source map mappings for the entire function
+        # (because we give func.offset as a lower bound). Exit the loop.
+        if not loc:
+          break
+        # Exit the loop only if a location info with source file information is
+        # found. If not, continue the search.
+        if loc.source:
+          break
+        addr -= 1
+
+      if loc and loc.source:
+        func_to_src[func_name] = normalize_path(loc.source)
+      else:
+        if not is_synthesized_func(func_name):
+          diagnostics.warn(f"No source file information found in the source map for function '{func_name}'")
+
+    for func_name, src in func_to_src.items():
+      if src not in src_to_funcs:
+        src_to_funcs[src] = []
+      src_to_funcs[src].append(func_name)
+
+  # Visit paths in the reverse sorting order, so that we can process inner paths
+  # first.
+  # e.g. If we have /a/b and /a/b/c, /a/b/c will come first, so we can assign
+  # functions contained in /a/b/c to it first and assign the maining functions
+  # to /a/b.
+  visited_funcs = set()
+  path_to_funcs = {}
+  for path in sorted(paths, reverse=True):
+    ppath = PurePath(path)
+    path_to_funcs[path] = []
+    for src, funcs in src_to_funcs.items():
+      psrc = PurePath(src)
+      if ppath == psrc or ppath in psrc.parents:
+        for func in funcs:
+          if func not in visited_funcs:
+            visited_funcs.add(func)
+            path_to_funcs[path].append(func)
+  return path_to_funcs
+
+
+def main():
+  args, forwarded_args = parse_args()
+  wasm = os.path.expanduser(args.wasm)
```

Comment:
Is there a particular reason you are using expanduser here?   We don't tend to that in most of our tools, but rely on the shell to do it for us.